common/lib/xmodule/xmodule/js/src/capa/display.js

```diff
@@ -377,406 +441,461 @@
+            statusElement = $(contents).find('.status');
+            labeledStatus = [];
+            for (i = 0, len = statusElement.length; i < len; i++) {
+                element = statusElement[i];
+                parentSection = $(element).closest('section');
+                addedStatus = false;
+                if (parentSection) {
+                    ariaLabel = parentSection.attr('aria-label');
+                    if (ariaLabel) {
+                        // Translators: This is only translated to allow for reordering of label and associated status.;
+                        template = gettext('{label}: {status}');
+                        labeledStatus.push(edx.StringUtils.interpolate(template, {
+                            label: ariaLabel,
                             status: $(element).text()
                         }));
-                        added_status = true;
+                        addedStatus = true;
                     }
                 }
-                if (!added_status) {
-                    labeled_status.push($(element).text());
+                if (!addedStatus) {
+                    labeledStatus.push($(element).text());
                 }
             }
-            return labeled_status;
+            return labeledStatus;
         };
 
-        Problem.prototype.reset = function () {
+        Problem.prototype.reset = function() {
             return this.disableAllButtonsWhileRunning(this.reset_internal, false);
         };
 
-        Problem.prototype.reset_internal = function () {
-            var _this = this;
+        Problem.prototype.reset_internal = function() {
+            var that = this;
             Logger.log('problem_reset', this.answers);
-            return $.postWithPrefix("" + this.url + "/problem_reset", {
+            return $.postWithPrefix('' + this.url + '/problem_reset', {
                 id: this.id
-            }, function (response) {
+            }, function(response) {
                 if (response.success) {
-                    _this.el.trigger('contentChanged', [_this.id, response.html]);
-                    _this.render(response.html, _this.scroll_to_problem_meta);
-                    _this.updateProgress(response);
+                    that.el.trigger('contentChanged', [that.id, response.html]);
+                    that.render(response.html, that.scroll_to_problem_meta);
+                    that.updateProgress(response);
                     return window.SR.readText(gettext('This problem has been reset.'));
                 } else {
-                    return _this.gentle_alert(response.msg);
+                    return that.gentle_alert(response.msg);
                 }
             });
         };
 
-        Problem.prototype.show = function () {
-            var _this = this;
+        // TODO this needs modification to deal with javascript responses; perhaps we
```

Comment:
I copied in the old TODOs from the CoffeeScript. I don't think there is value creating a story for this work (which may no longer actually be an issue... I'd have to dive in to understand the comment). I just thought it was better not to lose comments, in case they actually do have value.